### PR TITLE
PR: Scroll to the selected item after go next thumbnail

### PR DESCRIPTION
**Pull request by ok97465**
## Description of Changes

* [x] Wrote at least one-line docstrings (for any new functions)
* [x] Added unit test(s) covering the changes (if testable)
* [x] Included a screenshot (if affecting the UI)

### Result of master
---

* ThumbnailScrollBar is not scrolled after the go_next_thumbnail button is pressed.
Plots_current

### Result of PR
---

* ThumbnailScrollBar is scrolled after the go_next_thumbnail button is pressed.
Plots_fix

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: ok97465

### CODE REVIEW

File: spyder/plugins/plots/widgets/figurebrowser.py [hunk 1]
```diff
@@ -709,13 +709,35 @@ def go_previous_thumbnail(self):
             index = self._thumbnails.index(self.current_thumbnail) - 1
             index = index if index >= 0 else len(self._thumbnails) - 1
             self.set_current_index(index)
+            self.scroll_to_select_item(index)
```
Comment (goanpeca) [inline spyder/plugins/plots/widgets/figurebrowser.py]: Maybe just use `scroll_to_item`

File: spyder/plugins/plots/widgets/figurebrowser.py [hunk 2]
```diff
@@ -709,13 +709,35 @@ def go_previous_thumbnail(self):
...
 
     def go_next_thumbnail(self):
         """Select thumbnail next to the currently selected one."""
         if self.current_thumbnail is not None:
             index = self._thumbnails.index(self.current_thumbnail) + 1
             index = 0 if index >= len(self._thumbnails) else index
             self.set_current_index(index)
+            self.scroll_to_select_item(index)
+
+    def scroll_to_select_item(self, index):
```
Comment (goanpeca) [inline spyder/plugins/plots/widgets/figurebrowser.py]: I suggest we use `scroll_to_item` :-)

File: spyder/plugins/plots/widgets/figurebrowser.py [hunk 3]
```diff
@@ -709,13 +709,35 @@ def go_previous_thumbnail(self):
...
+
+        height_of_top_items = spacing_between_items
+        for i in range(index):
+            item = self.scene.itemAt(i)
+            height_of_top_items += item.sizeHint().height()
+            height_of_top_items += spacing_between_items
+
+        pos_scroll = height_of_top_items - height_view_excluding_item // 2
+
+        vsb = self.scrollarea.verticalScrollBar()
```
Comment (goanpeca) [inline spyder/plugins/plots/widgets/figurebrowser.py]: Is there a chance that there is no scroll bar at all if there are only a few or couple of items?

File: spyder/plugins/plots/widgets/tests/test_plots_widgets.py [hunk 4]
```diff
@@ -77,6 +77,8 @@ def add_figures_to_browser(figbrowser, nfig, tmpdir, fmt='image/png'):
         figs.append(create_figure(figname))
         figbrowser._handle_new_figure(figs[-1], fmt)
 
+    figbrowser.thumbnails_sb.scene.setVerticalSpacing(6)
```
Comment (goanpeca) [inline spyder/plugins/plots/widgets/tests/test_plots_widgets.py]: Just wondering, why was this needed?

File: spyder/plugins/plots/widgets/figurebrowser.py (hunk 2 above)
Comment (ccordoba12) [reply spyder/plugins/plots/widgets/figurebrowser.py]: +1 from me.

File: spyder/plugins/plots/widgets/tests/test_plots_widgets.py (hunk 4 above)
Comment (ok97465) [reply spyder/plugins/plots/widgets/tests/test_plots_widgets.py]: I added it  to find the cause of the test failure in macos. Currently it was removed.

File: spyder/plugins/plots/widgets/figurebrowser.py (hunk 3 above)
Comment (ok97465) [reply spyder/plugins/plots/widgets/figurebrowser.py]: ThumbnailScrollBar creates a verticalScrollBar in __init__ method.

https://github.com/spyder-ide/spyder/blob/8442cadd0f6fec887e0297f8dfe29fcfa17120fa/spyder/plugins/plots/widgets/figurebrowser.py#L537

**Comment by ccordoba12** [comment]
@goanpeca, something else to add?

File: spyder/plugins/plots/widgets/figurebrowser.py [hunk 5]
```diff
@@ -709,13 +709,35 @@ def go_previous_thumbnail(self):
...
             index = 0 if index >= len(self._thumbnails) else index
             self.set_current_index(index)
+            self.scroll_to_item(index)
+
+    def scroll_to_item(self, index):
+        """Scroll to the selected item of ThumbnailScrollBar."""
+        spacing_between_items = self.scene.verticalSpacing()
+        height_view = self.scrollarea.viewport().height()
+        height_item = self.scene.itemAt(index).sizeHint().height()
+        height_view_excluding_item = height_view - height_item
```
Comment (jnsebgosselin) [inline spyder/plugins/plots/widgets/figurebrowser.py]: The following three lines could be replaced by:
`height_view_excluding_item = max(0, height_view - height_item)`

File: spyder/plugins/plots/widgets/tests/test_plots_widgets.py [hunk 6]
```diff
@@ -235,6 +235,28 @@ def test_go_prev_next_thumbnail(figbrowser, tmpdir, fmt):
...
+        figbrowser.go_next_thumbnail()
+        qtbot.wait(500)
+
+    scene = figbrowser.thumbnails_sb.scene
+
+    spacing = scene.verticalSpacing()
+    height = scene.itemAt(0).sizeHint().height()
+    height_view = figbrowser.thumbnails_sb.scrollarea.viewport().height()
+
+    expected = (spacing * 5) + (height * 4) - ((height_view - height) // 2)
```
Comment (jnsebgosselin) [inline spyder/plugins/plots/widgets/tests/test_plots_widgets.py]: Would it be possible to replace the hard-coded 5 and 4 values by `nfig // 2` and `nfig//2 - 1)` ?

I think this would make the test easier to understand.

**Comment by jnsebgosselin** [comment]
I just left two minor optional suggestions, otherwise, LGTM.

Thank you very much for this @ok97465 !

File: spyder/plugins/plots/widgets/tests/test_plots_widgets.py (hunk 6 above)
Comment (goanpeca) [reply spyder/plugins/plots/widgets/tests/test_plots_widgets.py]: Indeed, I was not sure what those numbers meant

**Comment by goanpeca** [comment]
Thanks for the review @jnsebgosselin !

Thanks for the PR @ok97465
